plenum/server/replica.py

```diff
@@ -1639,17 +1653,21 @@ def dequeueCommits(self, viewNo: int, ppSeqNo: int):
         key = (viewNo, ppSeqNo)
         if key in self.commitsWaitingForPrepare:
             i = 0
-            # Keys of pending prepares that will be processed below
-            while self.commitsWaitingForPrepare[key]:
-                commit, sender = self.commitsWaitingForPrepare[
-                    key].popleft()
-                logger.debug("{} popping stashed COMMIT{}".format(self, key))
-                self.processCommit(commit, sender)
-                i += 1
-            self.commitsWaitingForPrepare.pop(key)
-            logger.debug("{} processed {} COMMITs waiting for PREPARE for"
-                         " view no {} and seq no {}".
-                         format(self, i, viewNo, ppSeqNo))
+            if self.has_prepared(key):
```

Comment:
The code becomes more readable if you process shorter case first (negative case here).
It's better to check that if we don't has prepared, log message and return:
```
if not self.has_prepared(key):
    log...
    return
....
```